Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Logging.rst

```diff
@@ -0,0 +1,157 @@
+.. _logging:
+
+=======
+Logging
+=======
+
+.. sectionauthor:: Alexander Berl <alexander@neos.io>
+
+In Flow logging is implemented according to the `PSR-3 Standard`_. This means you can use any logging facility that implements this interface.
+The concrete implementation can easily be configured with the :doc:`Object Management` of Flow. By default, Flow comes with an PSR-3 implementation
+in the ``Neos.Flow.Log`` package that can be configured with one or more storage backends and supports log rotation. The supported backends currently
+consist of a ``FileBackend``, ``(Ansi)ConsoleBackend``, ``JsonFileBackend`` and a ``NullBackend``.
+
+Default loggers
+===============
+
+By default Flow comes with four loggers, the so called "system logger", "security logger", "sql logger" and "i18n logger".
+As the names imply, the first is responsible for logging general system level messages and is used by default if an instance of the ``LoggerInterface`` is injected. The second for
+logging security related information. The SQL logger needs to be enabled first via ``Neos.Flow.persistence.doctrine.sqlLogger`` setting and will
+create a log of all database queries, so this can become a big performance penalty and should only be used for debugging purposes.
+Last but not least is the i18n logger which will log away all messages related to the translation framework, for example when the XLIFF translation
+sources are badly formatted.
+Under the hood they all use a ``FileBackend`` for storing the messages, but that can be configured differently via the settings keys ``Neos.Flow.log.psr3.Neos\Flow\Log\PsrLoggerFactory.*``.
+
+You can use these loggers for example if you extend the system via signals or AOP, or you are debugging
+your code. For that you inject a ``Psr\Log\LoggerInterface`` wherever you need it like this::
+
+	/**
+	 * @Flow\Inject(name="Neos.Flow:SystemLogger")
```

Comment:
The line above state, that injecting `LoggerInterface` by default injects the SystemLogger. Why is the name necessary here?

Reply:
Technically you can leave it for the `SystemLogger` but I think it's a bad idea having a default for this interface (debatable), it's more explicit to state that you expect the system logger at this point, but more importantly, I do think this better helps transport the concept of injecting via the virtual objects configuration, than having two examples for the default and non-default case.

Reply:
I see the point, that it is much clearer, what to expect here. But I use the system logger in 99,99 percent of the time (maybe because I don't like the concept of having different logfiles ... but you rarely see usages of other loggers, especially in 3rd party code). 

With these virtual objects, injects of this "standard" object is much more bulky now and lacks code completion. So it makes things more complicated in the majority of the cases.

Reply:
> lacks code completion

That's a valid point, but the only solution I see for this is if we reintroduce specific interfaces for the different framework loggers (`SystemLoggerInterface`, `SecurityLoggerInterface`, ...) that you can inject. Totally plausible and an alternative to the virtual objects. Maybe we should discuss this in the team first, if we want to go virtual objects or interfaces for loggers, as changing it again later would be bad?

Reply:
I would suggest that we use the `Psr\Log\LoggerInterface` as synonym for the System Logger by default (i.e. remove `name="Neos.Flow:SystemLogger"` from the examples).
IMO that doesn't really make it less explicit really. It just means that I want to use some default logger implementation and don't really care which one it is _at this point_.

Reply:
> that doesn't really make it less explicit really.

Guess that depends on the POV - and I can totally see your point now "don't really care which one".
We should document that clearly with examples for both - both is possible, so it's a decision (or non-decision) to make by the developer.

Reply:
Does https://github.com/neos/flow-development-collection/pull/2134/files#diff-a206dea4eb5ffcc6b7dc0386821eca9eR58-R65 suffice for you two?
I would leave the "name" attribute in this specific place, since it is meant to list the different (explicit) options, I removed it from the other example for ThrowableStorage below.

Reply:
So, according to the poll, we'll stick with virtual objects, even though they won't have autocomplete for the name. Regarding the usage of virtual object system logger injection in core, see my argument above and in the discuss thread. I'm open for counter-arguments though.